src/DataStructures/CMakeLists.txt

```diff
@@ -17,7 +17,11 @@ add_spectre_library(${LIBRARY} ${LIBRARY_SOURCES})
 
 target_link_libraries(
   ${LIBRARY}
-  INTERFACE ErrorHandling
+  PUBLIC
+  Blaze
+  Boost
+  ErrorHandling
+  Options
```

Comment:
Missing: `Utilities`

Reply:
If it's okay, the next thing I'm going to do is clean up all the dependencies in the `src` directory, and as part of that this will get fixed. Is that alright?

Reply:
Unfortunately the branch only compiles on my machine when I add `Utilities` here and then make linking with `DataStructures` public wherever `ConstantExpression.hpp` is being included (so basically everywhere). It's because `ConstantExpression.hpp` includes Blaze..

Reply:
What if you remove `IsVector` from `ConstantExpressions`? I don't see why that include is needed there...

Reply:
that works 👍 now it's only the libs that include `DenseMatrix` `DenseVector` or `Matrix` that need to link publicly with `DataStructures`. Haven't looked in more detail if that can be avoided, or what's if it's the same for `DataVector`, `Variables` and friends. I can do more detailed tests tomorrow. I do wonder, however, why you don't have these issues.

Reply:
Okay, I'm hopeful that those are actually correct (or can be corrected in a subsequent PR). I'll add a commit to the beginning that removes the problematic include. My guess is that everything that includes anything from DataStructures needs to link publicly against it, since we use our own data structures in the interfaces to libraries all the time (e.g. DataVector and Tensor)

If you're seeing the issue on macOS, not in the container, then it's just that Apple's linker is doing something different (unclear if more or less correct).